docker/pypi/wmagent-mariadb/mariadb-docker-run.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+
+### This script is to be used for running the Mariadb docker container at a VM
+### Its sole purpose is to set all the needed mount points from the Host VM and
+### forward all Mariadb runtime parameters to the Mariadb container entrypoint run.sh
+### It accepts only the set of parameters relevant to Mariadb's container run.sh
+### and no build dependent ones. The docker image tag to be searched for execution is
+### always `latest`.
+
+
+# NOTE: In the help call to the current scrit we only repeat the help and usage
+#       information for all the parameters accepted by run.sh.
+help(){
+    echo -e $*
+    cat <<EOF
+
+The script to be used for running a Mariadb docker container at a VM. The full set of arguments
+passed to the current script are to be forwarded to the Mariadb container entrypoint 'run.sh'
+
+Usage: mariadb-docker-run.sh [-t <team_name>] [-n <agent_number>] [-f <db_flavour>]
+
+    -p <pull_image>   Pull the image from registry.cern.ch
+    -t <mariadb_tag>  The Mariadb version/tag to be downloaded from registry.cern.ch [Default:latest]
+    -h <help>
+
+Example: ./mariadb-docker-run.sh -t 3.2.2
+
+EOF
+}
+
+usage(){
+    help $*
+    exit 1
+}
+
+PULL=false
+MDB_TAG=latest
+
+
+### Argument parsing:
+while getopts ":t:hp" opt; do
+    case ${opt} in
+        t) MDB_TAG=$OPTARG ;;
+        p) PULL=true ;;
+        h) help; exit $? ;;
+        : )
+            msg="Invalid Option: -$OPTARG requires an argument"
+            usage "$msg" ;;
+    esac
+done
+
+
+mariadbUser=cmst1
+mariadbOpts=" --user $mariadbUser"
+
+# This is the root at the host only, it may differ from the root inside the container.
+# NOTE: this may be parametriesed, so that the container can run on a different mount point.
+HOST_MOUNT_DIR=/data/dockerMount
```

Comment:
Well it is a matter of choice, but I find it much better to explicitly state that this is actually a mounted at the container. And it should be the same for all the three containers we run - WMAgent, MariaDB, CouchDB . If we decide to change it now, it will require changes in multiple places.

Reply:
We can of course parametrize it (as stated in the note above) and let it to the choice of the user who is doing the deployment. But as I said we need to parametrize this for the 3 containers, and we would have to rely on the user not to make the mistake during deployment to point the `WMAgent` container to one mount point at the host and the `MariaDB` to other and the `CouchDB` container to yet another.

Reply:
p.s. the `dockerMount` directory is completely stripped from the tree inside the container. Taking as an example  the `MariaDB.secrets` file:

From the host it will bee seen at:
```
cmst1@vocms0290:wmagent-mariadb $ ll /data/dockerMount/admin/mariadb/MariaDB.secrets
-rw-r--r--. 1 root root 37 Feb 24 10:05 /data/dockerMount/admin/mariadb/MariaDB.secrets
```  

from the container it would be:
```
(MariaDB-10.5) [cmst1@vocms0290:data]$ ll /data/admin/mariadb/MariaDB.secrets
-rw-r--r--. 1 root root 37 Feb 24 09:05 /data/admin/mariadb/MariaDB.secrets
```